crates/bevy_asset/src/io/file_asset_io.rs

```diff
@@ -27,12 +27,20 @@ pub struct FileAssetIo {
 }
 
 impl FileAssetIo {
-    pub fn new<P: AsRef<Path>>(path: P) -> Self {
-        FileAssetIo {
+    pub fn new<P: AsRef<Path>>(
+        path: P,
+        #[cfg(feature = "filesystem_watcher")] watch_for_changes: bool,
```

Comment:
Please unconditionally add this param even if you panic when it is set without `filesystem_watcher`. Cargo features are additive and thus enabling them must not break other crates that don't enable it.

Reply:
This is from #3643

Reply:
I see. Copied this review comment as https://github.com/bevyengine/bevy/pull/3643#discussion_r785283556.

Reply:
Fixed.